components/Icon.tsx

```diff
@@ -0,0 +1,47 @@
+import clsx from "clsx";
+
+type Props = {
+  type: "home" | "search";
+  className?: string;
+  size?: string;
```

Comment:
this should be more strict:
```suggestion
  size?: 'small' | 'medium';
```